pkg/api/types.go

```diff
@@ -1196,6 +1196,18 @@ func (p *Properties) GetAADAdminGroupID() string {
 	return ""
 }
 
+func (p *Properties) NeedsAuditdRules() bool {
```

Comment:
It's just because both node types derive cloud-init data from a common dictionary that gets populated to the ARM template. So if any one node pool (or control plane) opts into auditd, that data is needed in the template in order to be consumed by cloud-init anywhere.